File: tests/rosetta/transpiler/php/anonymous-recursion-1.php
<?php
function fib($n) {
  global $main;
  if ($n < 2) {
  return $n;
}
  $a = 0;
  $b = 1;
  $i = 1;
  while ($i < $n) {
  $t = $a + $b;
  $a = $b;
  $b = $t;
  $i = $i + 1;
};
  return $b;
}
function main() {
  global $fib;
  foreach ([0, 1, 2, 3, 4, 5, 10, 40, -1] as $n) {
  if ($n < 0) {
  echo "fib undefined for negative numbers", PHP_EOL;
} else {
  echo "fib " . json_encode($n, 1344) . " = " . json_encode(fib($n), 1344), PHP_EOL;
}
};
}
main();

File: tests/rosetta/transpiler/Java/anonymous-recursion-1.java
public class Main {

    static int fib(int n) {
        if (n < 2) {
            return n;
        }
        int a = 0;
        int b = 1;
        int i = 1;
        while (i < n) {
            int t = a + b;
            a = b;
            b = t;
            i = i + 1;
        }
        return b;
    }

    static void main() {
        for (var n : new int[]{0, 1, 2, 3, 4, 5, 10, 40, -1}) {
            if (n < 0) {
                System.out.println("fib undefined for negative numbers");
            } else {
                System.out.println(String.valueOf(String.valueOf("fib " + String.valueOf(n)) + " = ") + String.valueOf(fib(n)));
            }
        }
    }
    public static void main(String[] args) {
        main();
    }
}
